src/server.ts

```diff
@@ -11,22 +13,28 @@ export interface ServerInfo {
 
 export function start(): Promise<ServerInfo> {
     return new Promise((resolve, reject) => {
-        findJava((err, home) => {
+        findJava(async (err, home) => {
             if (err) {
                 reject(err);
             } else {
                 const serverLocation = path.resolve(__dirname, '..', '..', 'server');
                 const felix = path.join(serverLocation, 'bin', 'felix.jar');
                 const java = path.join(home, 'bin', 'java');
                 const process = cp.spawn(java, ['-jar', felix], { cwd: serverLocation });
```

Comment:
Selected port should be passed to the server through env variable or any other way it supports.

Reply:
See https://github.com/redhat-developer/rsp-server/issues/31 in rsp-server.